pkg/provider/beyondtrust/provider.go

```diff
@@ -0,0 +1,360 @@
+	}
+
+	clientSecret, err := loadConfigSecret(ctx, config.Auth.Clientsecret, kube, namespace)
+	if err != nil {
+		return nil, fmt.Errorf(errReturn, err)
+	}
+
+	if config.Auth.Certificate != nil && config.Auth.Certificatekey != nil {
+		loadedCertificate, err := loadConfigSecret(ctx, config.Auth.Certificate, kube, namespace)
+		if err != nil {
+			return nil, fmt.Errorf(errReturn, err)
+		}
+
+		certificate = loadedCertificate
+
+		loadedCertificateKey, err := loadConfigSecret(ctx, config.Auth.Certificatekey, kube, namespace)
+		if err != nil {
+			return nil, fmt.Errorf(errReturn, err)
+		}
+
+		certificateKey = loadedCertificateKey
+	}
+
+	// Create an instance of ValidationParams
+	params := utils.ValidationParams{
+		ClientID:                   clientID,
+		ClientSecret:               clientSecret,
+		ApiUrl:                     &apiURL,
+		ClientTimeOutInSeconds:     clientTimeOutInSeconds,
+		Separator:                  &separator,
+		VerifyCa:                   config.Server.VerifyCA,
+		Logger:                     logger,
+		Certificate:                certificate,
+		CertificateKey:             certificateKey,
+		RetryMaxElapsedTimeMinutes: &retryMaxElapsedTimeMinutes,
+		MaxFileSecretSizeBytes:     &maxFileSecretSizeBytes,
+	}
+
+	errorsInInputs := utils.ValidateInputs(params)
+
+	if errorsInInputs != nil {
+		return nil, fmt.Errorf(errReturn, errorsInInputs)
+	}
+
+	// creating a http client
+	httpClientObj, err := utils.GetHttpClient(clientTimeOutInSeconds, config.Server.VerifyCA, certificate, certificateKey, logger)
+
+	if err != nil {
+		return nil, fmt.Errorf(errReturn, err)
+	}
+
+	// instantiating authenticate obj, injecting httpClient object
+	authenticate, _ := auth.Authenticate(*httpClientObj, backoffDefinition, apiURL, clientID, clientSecret, logger, retryMaxElapsedTimeMinutes)
+
+	return &Provider{
+		apiURL:        config.Server.APIURL,
+		retrievaltype: config.Server.Retrievaltype,
+		authenticate:  *authenticate,
+		log:           *logger,
+		separator:     separator,
+	}, nil
+}
+
+func loadConfigSecret(ctx context.Context, ref *esv1beta1.BeyondTrustProviderSecretRef, kube client.Client, defaultNamespace string) (string, error) {
+	if ref.SecretRef == nil {
+		return ref.Value, nil
+	}
+
+	if err := validateSecretRef(ref); err != nil {
+		return "", err
+	}
+
+	namespace := defaultNamespace
+	if ref.SecretRef.Namespace != nil {
+		namespace = *ref.SecretRef.Namespace
+	}
+
+	ESOLogger.Info("using k8s secret", "name:", ref.SecretRef.Name, "namespace:", namespace)
+	objKey := client.ObjectKey{Namespace: namespace, Name: ref.SecretRef.Name}
+	secret := v1.Secret{}
+	err := kube.Get(ctx, objKey, &secret)
+	if err != nil {
+		return "", err
+	}
+
+	value, ok := secret.Data[ref.SecretRef.Key]
+	if !ok {
+		return "", fmt.Errorf(errNoSuchKeyFmt, ref.SecretRef.Key)
+	}
+
+	return string(value), nil
+}
+
+func validateSecretRef(ref *esv1beta1.BeyondTrustProviderSecretRef) error {
+	if ref.SecretRef != nil {
```

Comment:
Flip this if so that it returns immediately with the error and then proceeds normally.

Reply:
Done